sal-service/src/main/java/org/ow2/proactive/sal/service/service/NodeService.java

```diff
@@ -138,7 +149,11 @@ public synchronized Boolean addNodes(String sessionId, List<IaasDefinition> node
             LOGGER.info("Node source defined.");
 
             LOGGER.info("Trying to retrieve task: " + node.getTaskName());
-            Task task = repositoryService.getJob(jobId).findTask(node.getTaskName());
+            Task task = job.findTask(node.getTaskName());
+            if (task == null) {
+                LOGGER.error("Task [{}] does not exist.", node.getTaskName());
```

Comment:
does the task name contain the job id ? if not, is it possible to also print the associated job id ?

Reply:
No but I can add it.